File: e2e/node/tests/batch.test.ts
import { describe, it, expect } from "vitest";
import { scrape, createEngine } from "@kreuzberg/kreuzcrawl";

describe("batch", () => {
	it("scrape_batch_basic: Batch scrape of multiple URLs all succeeding", async () => {
		const engine = createEngine(null);
		await scrape(engine, "");
		// skipped: field 'batch.completed_count' not available on result type
		// skipped: field 'batch.failed_count' not available on result type
		// skipped: field 'batch.total_count' not available on result type
	});

	it("scrape_batch_partial_failure: Batch scrape with one URL failing returns partial results", async () => {
		const engine = createEngine(null);
		await scrape(engine, "");
		// skipped: field 'batch.completed_count' not available on result type
		// skipped: field 'batch.failed_count' not available on result type
		// skipped: field 'batch.total_count' not available on result type
	});

	it("scrape_batch_progress: Batch scrape results include specific URL", async () => {
		const engine = createEngine(null);
		await scrape(engine, "");
		// skipped: field 'batch.total_count' not available on result type
		// skipped: field 'batch.results' not available on result type
	});
});

File: e2e/elixir/test/batch_test.exs
# E2e tests for category: batch
defmodule E2e.BatchTest do
  use ExUnit.Case, async: true

  describe "scrape_batch_basic" do
    test "Batch scrape of multiple URLs all succeeding" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'batch.completed_count' not available on result type
      # skipped: field 'batch.failed_count' not available on result type
      # skipped: field 'batch.total_count' not available on result type
    end
  end

  describe "scrape_batch_partial_failure" do
    test "Batch scrape with one URL failing returns partial results" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'batch.completed_count' not available on result type
      # skipped: field 'batch.failed_count' not available on result type
      # skipped: field 'batch.total_count' not available on result type
    end
  end

  describe "scrape_batch_progress" do
    test "Batch scrape results include specific URL" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'batch.total_count' not available on result type
      # skipped: field 'batch.results' not available on result type
    end
  end
end
